torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2456,9 +2456,82 @@ Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const T
     return u_term + sigma_term + v_term + imag_term;
   }
 
+
   return u_term + sigma_term + v_term;
 }
 
+Tensor _linalg_svd_rank_restricted_helper_backward(
+    const std::vector<torch::autograd::Variable>& grads,
+    const Tensor& input,
+    const Tensor& U,
+    const Tensor& S,
+    const Tensor& Vh,
+    const Tensor& rank,
+    const Tensor& unique_rank
+) {
+  auto grad = at::empty_like(input);
+
+  // If empty tensor in batch dimensions
+  if (!input.numel()) {
+    return grad;
+  }
+
+  const auto& U_grad = grads[0];
+  const auto U_grad_defined = U_grad.defined();
+
+  const auto& S_grad = grads[1];
+  const auto S_grad_defined = S_grad.defined();
+
+  const auto& Vh_grad = grads[2];
+  const auto Vh_grad_defined = Vh_grad.defined();
```

Comment:
Better to simply unpack it in the derivatives.yaml as `grad_U, grad_S, grad_Vh`.